x-pack/plugins/observability_solution/inventory/public/hooks/use_kibana_query_settings.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EsQueryConfig } from '@kbn/es-query';
+import { SerializableRecord } from '@kbn/utility-types';
+import { useMemo } from 'react';
+import { UI_SETTINGS } from '@kbn/data-plugin/public';
+import { useUiSetting$ } from '@kbn/kibana-react-plugin/public';
+
+export const useKibanaQuerySettings = (): EsQueryConfig => {
```

Comment:
this is the 3rd plugin with this hook. Should we move it to `observability_shared`?